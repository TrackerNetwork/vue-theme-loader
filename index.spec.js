const readFile = require('./lib/io').readFile;
const loader = require('./index');

test('parseAttributes() should return the attributes from a given tag', () => {
  const result = loader.parseAttributes('<style lang="stylus" theme="brand">');
  expect(result).toEqual({
    lang: 'stylus',
    theme: 'brand'
  });
});

test('parseStyleTags() should return the style tags in the given source', done => {
  readFile('./fixtures/normalComponent.vue').then(src => {
    const result = loader.parseStyleTags(src);
    expect(result).toEqual(['<style test="true">']);
    done();
  });
});

test('parse() should returned parsed component', done => {
  readFile('./fixtures/styleOnly.vue').then(src => {
    const result = loader.parse(src);
    expect(result).toMatchSnapshot();
    done();
  });
});

test('loader() should do things', done => {
  readFile('./fixtures/css.vue').then(src => {
    const result = loader(src);
    done();
  });
});
