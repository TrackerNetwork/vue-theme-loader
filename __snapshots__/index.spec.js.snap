exports[`test parse() should returned parsed component 1`] = `
Object {
  "customBlocks": Array [],
  "script": null,
  "styles": Array [
    Object {
      "attrs": Object {
        "lang": "css",
      },
      "content": "",
      "end": 18,
      "lang": "css",
      "start": 18,
      "type": "style",
    },
  ],
  "template": null,
}
`;
